Graphics_Engine/Modify/ApplyGradientOptions.cs

```diff
@@ -0,0 +1,79 @@
+﻿
+
+using BH.Engine.Base;
+using BH.oM.Graphics;
+using BH.oM.Graphics.Colours;
+using BH.oM.Graphics.Enums;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Graphics
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****           Public Methods                  ****/
+        /***************************************************/
+
+        [Description("Sets up the properties of a GradientOptions object for usage.")]
+        [Input("gradientOptions", "GradientOptions object to modify.")]
+        [Input("allValues", "The values to set gradient auto range from. Optional if range is already set.")]
+        [Input("defaultGradient", "Sets which gradient to use as default if no gradient is already set. Defaults to BlueToRed.")]
+        [Output("gradientOptions", "A GradientOptions object which is ready for usage.")]
+        [PreviousVersion("4.2", "BH.Engine.Graphics.Query.AutoRange(BH.oM.Graphics.GradientOptions, System.Collections.Generic.IEnumerable<System.Double>)")]
+        [PreviousVersion("4.2", "BH.Engine.Graphics.Query.CenteringOptions(BH.oM.Graphics.GradientOptions)")]
+        [PreviousVersion("4.2", "BH.Engine.Graphics.Query.DefaultGradient(BH.oM.Graphics.GradientOptions, System.String)")]
+        public static GradientOptions ApplyGradientOptions(this GradientOptions gradientOptions, IEnumerable<double> allValues = null, string defaultGradient = "BlueToRed")
```

Comment:
Are there really 3 previous versions?

Reply:
Originally they were 3 separate methods, but they were made in a very particular way and essentially had to be used together in a certain order, so I merged them into one. I doubt anybody was calling them from the UI anyway, so versioning might not be necessary, but if they were it does actually make sense to version them all to this merged one since it's made in such a way as to not override manually set parameters.

Reply:
OK. Yes I have seem them in the StructuralEnginnering_Toolkit. Lets see of the versioning check likes that, I've not tried it before. If not the `MessageForDeleted` in the versioning json file may be an option if they are not that widely used.